### PR TITLE
Task/dr 575 crud quest rdm

### PR DESCRIPTION
**Pull request by stuartreid**
## Description

Update the application code in the service management lambdas to a functional CRUD standard for Questionnaire Responses Manager.

**Note**
The functional crud code is a copy/paste of healthcare-services-data-manager with minor edits to change the service name including mock unit tests.

This PR includes the merge of branch /task/DR-5751_Questionnaire_responses_tests for the integration tests added by Jake.

## Context

DR-575 - Backend Application CRUD - Questionnaire Responses Manager.
DR-5751_Questionnaire_responses_tests

## Type of changes

- [ ] Refactoring (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [x] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: tests/integration/config/config_dev.ini [hunk 1]
```diff
@@ -2,4 +2,9 @@
 locations_body = data_json/location_post_body.json
 healthcareservices_body = data_json/healthcareservice_post_body.json
 organisation_body = data_json/organisation_post_body.json
+<<<<<<< HEAD
+=======
```
Comment (tamaragoldschmidt) [inline tests/integration/config/config_dev.ini:6]: There are some rebase lines here that need removing

File: application/questionnaire-responses-data-manager/test/test_service.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        "name": mock_name,
+    }
+    return data
+
+
+@mock_dynamodb
+def test_get_record_by_id():
+    "Test get_record_by_id method"
+    table = create_mock_dynamodb()
+    healthcare_service_record = build_mock_questionnaire_responses_item()
```
Comment (timrickwood) [inline application/questionnaire-responses-data-manager/test/test_service.py:52]: questionnaire_response_record ?

File: application/questionnaire-responses-data-manager/test/test_service.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+@mock_dynamodb
+def test_add_record():
+    "Test add record method - eg used by POST"
+    table = create_mock_dynamodb()
+    # check record with id does not already exist
+    response = table.get_item(Key={"id": mock_post_id})
+    assert ("Item" in response) is False
+    # build record to add
+    healthcare_service_record = build_mock_questionnaire_responses_item()
```
Comment (timrickwood) [inline application/questionnaire-responses-data-manager/test/test_service.py:82]: questionnaire_response_record ?

File: application/questionnaire-responses-data-manager/test/test_service.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    assert response["Item"]["id"] == mock_post_id
+    assert response["Item"]["active"] == mock_status_update
+
+
+@mock_dynamodb
+def test_update_record():
+    "Test update record method - eg used by PUT"
+    table = create_mock_dynamodb()
+    # Create and add a mock item to the table
+    healthcare_service_record = build_mock_questionnaire_responses_item()
```
Comment (timrickwood) [inline application/questionnaire-responses-data-manager/test/test_service.py:96]: questionnaire_response_record ?

File: application/questionnaire-responses-data-manager/test/test_service.py [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+    response = table.get_item(Key={"id": mock_id})
+    assert response["Item"]["active"] == mock_status_update
+    assert response["Item"]["name"] == mock_updated_name
+
+
+@mock_dynamodb
+def test_delete_record_by_id():
+    "Test delete_record method first adding and then checking it exists"
+    table = create_mock_dynamodb()
+    healthcare_service_record = build_mock_questionnaire_responses_item()
```
Comment (timrickwood) [inline application/questionnaire-responses-data-manager/test/test_service.py:116]: questionnaire_response_record ?

**Review by timrickwood — APPROVED** [review]

**Review by tamaragoldschmidt — APPROVED** [review]

**Pull request merged**